### DISCUSSION TITLE
How to get model field types in v2?

### DISCUSSION THREAD

**Discussion by JanEricNitschke** [pydantic/pydantic#10915] [Question] [answered]
I am trying to write a generic class that takes a pydantic model type, however the model can only have string fields. So i am trying to verify this at runtime.

I looked and found [this](https://stackoverflow.com/a/75827637/7895542) answer, but it does not seem to work in v2 as the `FieldInfo` type that is returned as the values of the dict from `model_info` does not have a `type_` property.

**Comment by EDohmen** [comment, marked as answer]
Are you looking for the model_fields dict with field.annotation (see example below)?

```python
from pydantic import BaseModel


class MyModel(BaseModel):
    a: int
    b: str
    c: list[float]


if __name__ == "__main__":
    for field_name, field in MyModel.model_fields.items():
        print(field_name, field.annotation, type(field.annotation))
```
output:
```bash
a <class 'int'> <class 'type'>
b <class 'str'> <class 'type'>
c list[float] <class 'types.GenericAlias'>
```

**Reply by JanEricNitschke** [reply]
`field.annotation` is exactly what i need.

```
for field in MyModel.model_fields.values():
    assert(field.annotation is str)
```
